src/traces/scatter3d/attributes.js

```diff
@@ -160,7 +160,7 @@ var attrs = module.exports = overrideAll({
         colorAttributes('marker')
     ),
 
-    textposition: extendFlat({}, scatterAttrs.textposition, {dflt: 'top center', arrayOk: false}),
+    textposition: extendFlat({}, scatterAttrs.textposition, {dflt: 'top center', arrayOk: true}),
```

Comment:
Making this line

```js
textposition: extendFlat({}, scatterAttrs.textposition, {dflt: 'top center'}),
```

would suffice as `textposition` in the scatter attributes is already `arrayOk`:

https://github.com/plotly/plotly.js/blob/fe2d8d7b64c397044a4577e0dd61a001e4366584/src/traces/scatter/attributes.js#L520-L535